types/document.d.ts

```diff
@@ -207,7 +209,7 @@ declare module 'mongoose' {
     populated(path: string): any;
 
     /** Sends a replaceOne command with this document `_id` as the query selector. */
-    replaceOne(replacement?: AnyObject, options?: QueryOptions | null): Query<any, this>;
+    replaceOne(replacement?: Partial<Omit<DocType, '_id'>>, options?: QueryOptions | null): Query<any, this>;
```

Comment:
This is incorrect. 1) `AnyObject` is more correct, because Mongoose will run casting to handle any types that don't match, 2) `Partial<>` is incorrect because required fields are still required, 3) omitting `_id` is incorrect: you can specify `_id` as long as the replacement has the same `_id` as the document, but there's no way to know whether the `_id` is the same at compile time.